// web-elements/dist/collection/components/checkbox-group/checkbox-group.js
import { Host, h } from "@stencil/core";
export class CheckboxGroup {
    constructor() {
        this.children = [];
    }
    checkboxCallbackHandler(prop) {
        const { value, checked } = prop.detail;
        this.children.forEach((child) => {
            if (child['value'] == value) {
                child['checked'] = checked;
            }
        });
        this.checkboxGroupCallback.emit({ value: this.value, children: this.children });
    }
    componentDidLoad() {
        const items = this.el.querySelectorAll(':scope > we-checkbox');
        items.forEach((i) => {
            const child = {};
            const value = i.getAttribute('value');
            child['value'] = value;
            const checked = i.getAttribute('checked');
            child['checked'] = checked === 'true' ? true : false;
            this.children.push(child);
        });
    }
    render() {
        return (h(Host, null,
            h("slot", null)));
    }
    static get is() { return "we-checkbox-group"; }
    static get encapsulation() { return "shadow"; }
    static get properties() { return {
        "value": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "string",
                "resolved": "string",
                "references": {}
            },
            "required": true,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Value that identify this checkbox group"
            },
            "attribute": "value",
            "reflect": false
        }
    }; }
    static get states() { return {
        "children": {}
    }; }
    static get events() { return [{
            "method": "checkboxGroupCallback",
            "name": "checkboxGroupCallback",
            "bubbles": true,
            "cancelable": true,
            "composed": true,
            "docs": {
                "tags": [],
                "text": "Event triggered when a checkbox inside change its state that returning the value of checkbox group and the value of checkbox changed"
            },
            "complexType": {
                "original": "CheckboxGroupValue",
                "resolved": "CheckboxGroupValue",
                "references": {
                    "CheckboxGroupValue": {
                        "location": "local"
                    }
                }
            }
        }]; }
    static get elementRef() { return "el"; }
    static get listeners() { return [{
            "name": "checkboxCallback",
            "method": "checkboxCallbackHandler",
            "target": undefined,
            "capture": false,
            "passive": false
        }]; }
}
