// web-elements/dist/esm/we-checkbox-group.entry.js
import { r as registerInstance, c as createEvent, h, H as Host, g as getElement } from './index-57b84139.js';

const CheckboxGroup = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.children = [];
        this.checkboxGroupCallback = createEvent(this, "checkboxGroupCallback", 7);
    }
    checkboxCallbackHandler(prop) {
        const { value, checked } = prop.detail;
        this.children.forEach((child) => {
            if (child['value'] == value) {
                child['checked'] = checked;
            }
        });
        this.checkboxGroupCallback.emit({ value: this.value, children: this.children });
    }
    componentDidLoad() {
        const items = this.el.querySelectorAll(':scope > we-checkbox');
        items.forEach((i) => {
            const child = {};
            const value = i.getAttribute('value');
            child['value'] = value;
            const checked = i.getAttribute('checked');
            child['checked'] = checked === 'true' ? true : false;
            this.children.push(child);
        });
    }
    render() {
        return (h(Host, null, h("slot", null)));
    }
    get el() { return getElement(this); }
};

export { CheckboxGroup as we_checkbox_group };
