// web-elements/dist/esm/we-radio-group.entry.js
import { r as registerInstance, c as createEvent, h, H as Host, g as getElement } from './index-57b84139.js';

const RadioGroup = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.children = [];
        this.radioGroupCallback = createEvent(this, "radioGroupCallback", 7);
    }
    radioCallbackHandler(prop) {
        const { value } = prop.detail;
        this.children.forEach((child) => {
            if (child['value'] == value)
                child['checked'] = true;
            else
                child['checked'] = false;
        });
        const radios = this.el.querySelectorAll(':scope > we-radio');
        radios.forEach((radio) => {
            const valueAttribute = radio.getAttribute('value');
            if (value != valueAttribute)
                radio.setAttribute('checked', 'false');
            else
                radio.setAttribute('checked', 'true');
        });
        this.radioGroupCallback.emit({ value: this.value, children: this.children });
    }
    componentDidLoad() {
        const radios = this.el.querySelectorAll(':scope > we-radio');
        radios.forEach((radio) => {
            const child = {};
            const value = radio.getAttribute('value');
            child['value'] = value;
            const checked = radio.getAttribute('checked');
            child['checked'] = checked === 'true' ? true : false;
            this.children.push(child);
        });
    }
    render() {
        return (h(Host, null, h("slot", null)));
    }
    get el() { return getElement(this); }
};

export { RadioGroup as we_radio_group };
