// web-elements/dist/collection/components/chips-group/chips-group.js
import { Host, h } from "@stencil/core";
export class ChipsGroup {
    constructor() {
        this.children = [];
    }
    removeCallbackHandler(prop) {
        const { value } = prop.detail;
        const getRemovedIndex = this.children.findIndex((item) => item.value === value);
        if (getRemovedIndex >= 0) {
            this.children.splice(getRemovedIndex, 1);
        }
        this.chipsGroupCallback.emit({ value: this.value, children: this.children });
    }
    selectCallbackHandler(prop) {
        const { value, isSelected } = prop.detail;
        this.children.forEach((child) => {
            if (child['value'] === value)
                child['isSelected'] = isSelected;
        });
        this.chipsGroupCallback.emit({ value: this.value, children: this.children });
    }
    componentDidLoad() {
        const items = this.el.querySelectorAll(':scope > we-chips');
        items.forEach((i) => {
            const child = {};
            const value = i.getAttribute('value');
            child['value'] = value;
            child['isSelected'] = false;
            this.children.push(child);
        });
    }
    render() {
        return (h(Host, null,
            h("slot", null)));
    }
    static get is() { return "we-chips-group"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() { return {
        "$": ["chips-group.scss"]
    }; }
    static get styleUrls() { return {
        "$": ["chips-group.css"]
    }; }
    static get properties() { return {
        "value": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "string",
                "resolved": "string",
                "references": {}
            },
            "required": true,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Value that identify this chips group"
            },
            "attribute": "value",
            "reflect": false
        }
    }; }
    static get states() { return {
        "children": {}
    }; }
    static get events() { return [{
            "method": "chipsGroupCallback",
            "name": "chipsGroupCallback",
            "bubbles": true,
            "cancelable": true,
            "composed": true,
            "docs": {
                "tags": [],
                "text": ""
            },
            "complexType": {
                "original": "ChipsGroupValue",
                "resolved": "ChipsGroupValue",
                "references": {
                    "ChipsGroupValue": {
                        "location": "local"
                    }
                }
            }
        }]; }
    static get elementRef() { return "el"; }
    static get listeners() { return [{
            "name": "removeChipsCallback",
            "method": "removeCallbackHandler",
            "target": undefined,
            "capture": false,
            "passive": false
        }, {
            "name": "selectChipsCallback",
            "method": "selectCallbackHandler",
            "target": undefined,
            "capture": false,
            "passive": false
        }]; }
}
