// web-elements/dist/collection/components/switch-group/switch-group.js
import { Host, h } from "@stencil/core";
export class SwitchGroup {
    constructor() {
        this.children = [];
    }
    changeSwitchCallbackHandler(prop) {
        const { value, checked } = prop.detail;
        this.children.forEach((child) => {
            if (child['value'] == value) {
                child['checked'] = checked;
            }
        });
        this.switchGroupCallback.emit({ value: this.value, children: this.children });
    }
    componentDidLoad() {
        const items = this.el.querySelectorAll(':scope > we-switch');
        items.forEach((i) => {
            const child = {};
            const value = i.getAttribute('value');
            child['value'] = value;
            const checked = i.getAttribute('checked');
            child['checked'] = checked === 'false' ? false : true;
            this.children.push(child);
        });
    }
    render() {
        return (h(Host, null,
            h("slot", null)));
    }
    static get is() { return "we-switch-group"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() { return {
        "$": ["switch-group.scss"]
    }; }
    static get styleUrls() { return {
        "$": ["switch-group.css"]
    }; }
    static get properties() { return {
        "value": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "string",
                "resolved": "string",
                "references": {}
            },
            "required": true,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Name that identify this switch group"
            },
            "attribute": "value",
            "reflect": false
        }
    }; }
    static get states() { return {
        "children": {}
    }; }
    static get events() { return [{
            "method": "switchGroupCallback",
            "name": "switchGroupCallback",
            "bubbles": true,
            "cancelable": true,
            "composed": true,
            "docs": {
                "tags": [],
                "text": "Function called when a switch inside change it's state"
            },
            "complexType": {
                "original": "SwitchGroupValue",
                "resolved": "SwitchGroupValue",
                "references": {
                    "SwitchGroupValue": {
                        "location": "local"
                    }
                }
            }
        }]; }
    static get elementRef() { return "el"; }
    static get listeners() { return [{
            "name": "switchCallback",
            "method": "changeSwitchCallbackHandler",
            "target": undefined,
            "capture": false,
            "passive": false
        }]; }
}
