// web-elements/dist/esm-es5/we-chips-group.entry.js
import { r as registerInstance, c as createEvent, h, H as Host, g as getElement } from './index-57b84139.js';
var ChipsGroup = /** @class */ (function () {
    function ChipsGroup(hostRef) {
        registerInstance(this, hostRef);
        this.children = [];
        this.chipsGroupCallback = createEvent(this, "chipsGroupCallback", 7);
    }
    ChipsGroup.prototype.removeCallbackHandler = function (prop) {
        var value = prop.detail.value;
        var getRemovedIndex = this.children.findIndex(function (item) { return item.value === value; });
        if (getRemovedIndex >= 0) {
            this.children.splice(getRemovedIndex, 1);
        }
        this.chipsGroupCallback.emit({ value: this.value, children: this.children });
    };
    ChipsGroup.prototype.selectCallbackHandler = function (prop) {
        var _a = prop.detail, value = _a.value, isSelected = _a.isSelected;
        this.children.forEach(function (child) {
            if (child['value'] === value)
                child['isSelected'] = isSelected;
        });
        this.chipsGroupCallback.emit({ value: this.value, children: this.children });
    };
    ChipsGroup.prototype.componentDidLoad = function () {
        var _this = this;
        var items = this.el.querySelectorAll(':scope > we-chips');
        items.forEach(function (i) {
            var child = {};
            var value = i.getAttribute('value');
            child['value'] = value;
            child['isSelected'] = false;
            _this.children.push(child);
        });
    };
    ChipsGroup.prototype.render = function () {
        return (h(Host, null, h("slot", null)));
    };
    Object.defineProperty(ChipsGroup.prototype, "el", {
        get: function () { return getElement(this); },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ChipsGroup, "style", {
        get: function () { return ":host{display:var(--display, flex);-ms-flex-pack:var(--justify-content, unset);justify-content:var(--justify-content, unset);-ms-flex-align:var(--align-items, unset);align-items:var(--align-items, unset);-ms-flex-direction:var(--flex-direction, row);flex-direction:var(--flex-direction, row)}"; },
        enumerable: true,
        configurable: true
    });
    return ChipsGroup;
}());
export { ChipsGroup as we_chips_group };
