// web-elements/dist/esm-es5/we-chips.entry.js
import { r as registerInstance, c as createEvent, h, H as Host } from './index-57b84139.js';
var Chips = /** @class */ (function () {
    function Chips(hostRef) {
        registerInstance(this, hostRef);
        this.isVisible = true;
        this.removeChipsCallback = createEvent(this, "removeChipsCallback", 7);
        this.selectChipsCallback = createEvent(this, "selectChipsCallback", 7);
    }
    Chips.prototype.handleRemoveChips = function (side) {
        if ((this.removeRight && side === 'right') || (this.removeLeft && side === 'left')) {
            this.isVisible = !this.isVisible;
            this.removeChipsCallback.emit({ value: this.value });
        }
    };
    Chips.prototype.handleSelectedChips = function () {
        if (this.isVisible) {
            if (this.isSelectable) {
                this.isSelected = !this.isSelected;
            }
        }
        else if (this.isSelectable) {
            this.isSelected = false;
        }
        this.selectChipsCallback.emit({ value: this.value, isSelected: this.isSelected });
    };
    Chips.prototype.render = function () {
        var _this = this;
        if (this.isVisible)
            return (h(Host, null, h("div", { class: 'chips ' + (this.isSelected && 'selected'), onClick: function () { return _this.handleSelectedChips(); } }, this.srcImgLeft && (h("img", { class: "imageLeft", src: this.srcImgLeft, onClick: function () { return _this.handleRemoveChips('left'); } })), this.label && h("label", null, this.label), this.srcImgRight && (h("img", { class: "imageRight", src: this.srcImgRight, onClick: function () { return _this.handleRemoveChips('right'); } })))));
    };
    Object.defineProperty(Chips, "style", {
        get: function () { return ".chips{background:var(--background, #e5e5e5);border-radius:var(--border-radius, 0.5em);border-width:var(--border-width, 0.05em);border-color:var(--border-color, unset);width:var(--width, auto);height:var(--heigth, auto);padding:var(--padding, 0.25rem);margin:var(--margin, 0);display:var(--display, inline-block)}.chips.selected{background:var(--selected-background, #41a9c0)}.imageLeft{cursor:var(--cursor, auto);padding:var(--imageLeft--padding, 0);margin:var(--imageLeft--margin, 0 0.5rem);width:var(--imageLeft--width, auto);height:var(--imageLeft--heigth, auto);border-radius:var(--imageLeft--border-radius, 0em)}label{color:var(--color, black);font-size:var(--font-size, 12pt);font-weight:var(--font-weight, normal);text-transform:var(--text-transform, unset);text-align:var(--text-align, left);padding:var(--label--padding, 0);margin:var(--label--margin, 0 0.5rem);-webkit-text-decoration:var(--text-decoration, unset);text-decoration:var(--text-decoration, unset);pointer-events:none}.imageRight{cursor:var(--cursor, pointer);padding:var(--imageRight--padding, 0);margin:var(--imageRight--margin, 0 0.5rem);width:var(--imageRight--width, auto);height:var(--imageRight--heigth, auto);border-radius:var(--imageRight--border-radius, 0em)}"; },
        enumerable: true,
        configurable: true
    });
    return Chips;
}());
export { Chips as we_chips };
