// web-elements/dist/esm-es5/we-modal.entry.js
import { r as registerInstance, c as createEvent, h, H as Host } from './index-57b84139.js';
var Modal = /** @class */ (function () {
    function Modal(hostRef) {
        registerInstance(this, hostRef);
        this.modalCallback = createEvent(this, "modalCallback", 7);
    }
    Modal.prototype.handleClose = function (event) {
        this.isVisible = false;
        this.modalCallback.emit(event);
    };
    Modal.prototype.render = function () {
        var classVisible = this.isVisible ? 'visible' : '';
        return (h(Host, null, h("div", { class: classVisible + ' back', onClick: this.handleClose.bind(this) }, h("div", { class: "modal" }, h("slot", null)))));
    };
    Object.defineProperty(Modal, "style", {
        get: function () { return ":host{display:block}.back{display:none;width:100vw;height:100vh;background:var(--back--background, rgba(128, 128, 128, 0.35));position:fixed;-ms-flex-pack:var(--back--justify-content, center);justify-content:var(--back--justify-content, center);-ms-flex-align:var(--back--align-items, center);align-items:var(--back--align-items, center)}.back.visible{display:-ms-flexbox;display:flex}.modal{display:var(--display, flex);background:var(--background, white);width:var(--width, 12.5em);height:var(--height, 18.75em);padding:var(--padding, 0.625em);border-radius:var(--border-radius, 0.625em);border:var(--border, none);-ms-flex-pack:var(--justify-content, center);justify-content:var(--justify-content, center);-ms-flex-align:var(--align-items, unset);align-items:var(--align-items, unset)}"; },
        enumerable: true,
        configurable: true
    });
    return Modal;
}());
export { Modal as we_modal };
