// web-elements/dist/esm-es5/we-switch-group.entry.js
import { r as registerInstance, c as createEvent, h, H as Host, g as getElement } from './index-57b84139.js';
var SwitchGroup = /** @class */ (function () {
    function SwitchGroup(hostRef) {
        registerInstance(this, hostRef);
        this.children = [];
        this.switchGroupCallback = createEvent(this, "switchGroupCallback", 7);
    }
    SwitchGroup.prototype.changeSwitchCallbackHandler = function (prop) {
        var _a = prop.detail, value = _a.value, checked = _a.checked;
        this.children.forEach(function (child) {
            if (child['value'] == value) {
                child['checked'] = checked;
            }
        });
        this.switchGroupCallback.emit({ value: this.value, children: this.children });
    };
    SwitchGroup.prototype.componentDidLoad = function () {
        var _this = this;
        var items = this.el.querySelectorAll(':scope > we-switch');
        items.forEach(function (i) {
            var child = {};
            var value = i.getAttribute('value');
            child['value'] = value;
            var checked = i.getAttribute('checked');
            child['checked'] = checked === 'false' ? false : true;
            _this.children.push(child);
        });
    };
    SwitchGroup.prototype.render = function () {
        return (h(Host, null, h("slot", null)));
    };
    Object.defineProperty(SwitchGroup.prototype, "el", {
        get: function () { return getElement(this); },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SwitchGroup, "style", {
        get: function () { return ":host{display:var(--display, flex);-ms-flex-pack:var(--justify-content, unset);justify-content:var(--justify-content, unset);-ms-flex-align:var(--align-items, unset);align-items:var(--align-items, unset);-ms-flex-direction:var(--flex-direction, row);flex-direction:var(--flex-direction, row)}"; },
        enumerable: true,
        configurable: true
    });
    return SwitchGroup;
}());
export { SwitchGroup as we_switch_group };
