// web-elements/dist/esm-es5/we-slider.entry.js
import { r as registerInstance, h, H as Host } from './index-57b84139.js';
import { p as parseFunction } from './index-fb1e05d5.js';
var WeSlider = /** @class */ (function () {
    function WeSlider(hostRef) {
        registerInstance(this, hostRef);
        /** Identify if slider is disabled or not */
        this.disabled = false;
    }
    WeSlider.prototype.handleChange = function (event) {
        if (this.changeCallback) {
            this.changeCallback = parseFunction(this.changeCallback);
            this.changeCallback({ name: this.name, value: event.target.value });
        }
    };
    WeSlider.prototype.render = function () {
        return (h(Host, null, h("label", { htmlFor: "weslider" }, h("slot", null)), h("input", { type: "range", id: "weslider", name: this.name, min: this.min, max: this.max, value: this.value, disabled: this.disabled, onChange: this.handleChange.bind(this) })));
    };
    Object.defineProperty(WeSlider, "style", {
        get: function () { return ":host{display:block}label{display:-ms-flexbox;display:flex;margin:var(--label-margin, 0.2em);padding:var(--label-padding, 0.2em)}input[type=range]{-webkit-appearance:none;height:var(--slider-height, 5px);width:var(--slider-width, 100%);border-radius:var(--slider-border-radius, 5px);background:var(--slider-background-color, #d7dcdf);outline:none;padding:var(--slider-padding, 0);margin:var(--slider-margin, 0)}input[type=range]::-webkit-slider-thumb{-webkit-appearance:none;width:var(--slider-thumb-width, 2em);height:var(--slider-thumb-height, 2em);border-radius:var(--slider-thumb-radius, 50%);background:var(--slider-thumb-color, #3f51b5);cursor:pointer}input[type=range]::-moz-range-thumb{width:var(--slider-thumb-width, 2em);height:var(--slider-thumb-height, 2em);background:var(--slider-thumb-color, #3f51b5);border-radius:var(--slider-thumb-radius, 50%);cursor:pointer}input[type=range]:disabled::-webkit-slider-thumb{background:var(--slider-disabled-color, #aaa)}input[type=range]:disabled::-moz-range-thumb{background:var(--slider-disabled-color, #aaa)}"; },
        enumerable: true,
        configurable: true
    });
    return WeSlider;
}());
export { WeSlider as we_slider };
