// web-elements/dist/esm-es5/we-tab.entry.js
import { r as registerInstance, c as createEvent, h, H as Host } from './index-57b84139.js';
import { g as generateUniqueId } from './index-fb1e05d5.js';
var Tab = /** @class */ (function () {
    function Tab(hostRef) {
        registerInstance(this, hostRef);
        this.valueId = generateUniqueId();
        this.tabCallback = createEvent(this, "tabCallback", 7);
    }
    Tab.prototype.handleChange = function (event) {
        this.tabCallback.emit(event.target.value);
    };
    Tab.prototype.render = function () {
        return (h(Host, { "data-id": this.valueId }, h("input", { type: "radio", id: "wetab", name: "tab", checked: this.enabled, value: this.valueId, onChange: this.handleChange.bind(this) }), h("label", { htmlFor: "wetab" }, h("slot", { name: "header" })), h("slot", { name: "content" })));
    };
    Object.defineProperty(Tab, "style", {
        get: function () { return ":host input{display:none}:host input:checked+label{background:var(--enabled-tab-color, #d7dcdf)}:host label{display:inline-block;font-size:var(--label-font-size, 1rem);height:var(--label-height, 2.25rem);line-height:var(--label-line-height, 2.25rem);width:100%;text-align:var(--label-text-align, center);background:var(--label-background-color, #ffffff);color:var(--label-color, #212121);position:relative;-webkit-transition:0.25s background ease;transition:0.25s background ease;cursor:pointer}:host label::after{content:\"\";height:var(--hover-height, 0.125rem);width:100%;position:absolute;display:block;background:var(--hover-color, #ccc);bottom:0;opacity:0;left:0;-webkit-transition:0.25s ease;transition:0.25s ease}:host label:hover::after{opacity:1}:host slot[name=content]{display:none}"; },
        enumerable: true,
        configurable: true
    });
    return Tab;
}());
export { Tab as we_tab };
