// web-elements/dist/esm-es5/we-tab-group.entry.js
import { r as registerInstance, h, H as Host, g as getElement } from './index-57b84139.js';
var TabGroup = /** @class */ (function () {
    function TabGroup(hostRef) {
        registerInstance(this, hostRef);
        this.contentSlot = document.createElement('div');
        this.contentSlot.slot = 'tab-content';
        this.tabs = this.el.querySelectorAll('we-tab');
    }
    TabGroup.prototype.tabCallbackHandler = function (event) {
        var _this = this;
        var value = event.detail;
        var line = this.el.shadowRoot.querySelector('.line');
        this.tabs.forEach(function (tab, tabPosition) {
            tab.style.width = 100 / _this.tabs.length + "%";
            if (tab.getAttribute('data-id') === value) {
                var left = tabPosition * (100 / _this.tabs.length);
                var currentContent = tab.querySelector('[slot="content"]');
                _this.contentSlot.innerHTML = currentContent.innerHTML;
                _this.el.appendChild(_this.contentSlot);
                tab.setAttribute('enabled', 'true');
                line.style.left = left + "%";
            }
            else {
                tab.setAttribute('enabled', 'false');
            }
        });
    };
    TabGroup.prototype.initTabs = function () {
        var _this = this;
        var line = this.el.shadowRoot.querySelector('.line');
        var getTabEnabledPosition = 0;
        this.tabs.forEach(function (tab, position) {
            if (tab.getAttribute('enabled') === 'true') {
                getTabEnabledPosition = position;
            }
            tab.style.width = 100 / _this.tabs.length + "%";
        });
        var defaultContent = this.tabs[getTabEnabledPosition].querySelector('[slot="content"]');
        var left = getTabEnabledPosition * (100 / this.tabs.length);
        line.style.left = left + "%";
        this.contentSlot.innerHTML = defaultContent.innerHTML;
        line.style.width = 100 / this.tabs.length + "%";
        this.el.appendChild(this.contentSlot);
    };
    TabGroup.prototype.componentDidRender = function () {
        this.initTabs();
    };
    TabGroup.prototype.render = function () {
        return (h(Host, null, h("header", null, h("slot", null), h("div", { class: "line" })), h("slot", { name: "tab-content" })));
    };
    Object.defineProperty(TabGroup.prototype, "el", {
        get: function () { return getElement(this); },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(TabGroup, "style", {
        get: function () { return ":host header{display:-ms-flexbox;display:flex;position:relative}:host header .line{position:absolute;z-index:10;height:var(--enabled-height, 0.125rem);background:var(--enabled-background-color, #3f51b5);top:var(--enabled-top, 2.125rem);left:0;-webkit-transition:0.25s ease;transition:0.25s ease}"; },
        enumerable: true,
        configurable: true
    });
    return TabGroup;
}());
export { TabGroup as we_tab_group };
