// web-elements/dist/lib/generated/web-elements-react/src/components.ts
/* eslint-disable */
/* tslint:disable */
/* auto-generated react proxies */
import { createReactComponent } from './react-component-lib';

import { JSX } from 'web-elements';



export const WeAccordion = /*@__PURE__*/createReactComponent<JSX.WeAccordion, HTMLWeAccordionElement>('we-accordion');
export const WeAccordionGroup = /*@__PURE__*/createReactComponent<JSX.WeAccordionGroup, HTMLWeAccordionGroupElement>('we-accordion-group');
export const WeButton = /*@__PURE__*/createReactComponent<JSX.WeButton, HTMLWeButtonElement>('we-button');
export const WeCard = /*@__PURE__*/createReactComponent<JSX.WeCard, HTMLWeCardElement>('we-card');
export const WeCheckbox = /*@__PURE__*/createReactComponent<JSX.WeCheckbox, HTMLWeCheckboxElement>('we-checkbox');
export const WeCheckboxGroup = /*@__PURE__*/createReactComponent<JSX.WeCheckboxGroup, HTMLWeCheckboxGroupElement>('we-checkbox-group');
export const WeChips = /*@__PURE__*/createReactComponent<JSX.WeChips, HTMLWeChipsElement>('we-chips');
export const WeChipsGroup = /*@__PURE__*/createReactComponent<JSX.WeChipsGroup, HTMLWeChipsGroupElement>('we-chips-group');
export const WeDivider = /*@__PURE__*/createReactComponent<JSX.WeDivider, HTMLWeDividerElement>('we-divider');
export const WeDropdownGroup = /*@__PURE__*/createReactComponent<JSX.WeDropdownGroup, HTMLWeDropdownGroupElement>('we-dropdown-group');
export const WeDropdownItem = /*@__PURE__*/createReactComponent<JSX.WeDropdownItem, HTMLWeDropdownItemElement>('we-dropdown-item');
export const WeList = /*@__PURE__*/createReactComponent<JSX.WeList, HTMLWeListElement>('we-list');
export const WeListItem = /*@__PURE__*/createReactComponent<JSX.WeListItem, HTMLWeListItemElement>('we-list-item');
export const WeModal = /*@__PURE__*/createReactComponent<JSX.WeModal, HTMLWeModalElement>('we-modal');
export const WeNavigationDrawer = /*@__PURE__*/createReactComponent<JSX.WeNavigationDrawer, HTMLWeNavigationDrawerElement>('we-navigation-drawer');
export const WeRadio = /*@__PURE__*/createReactComponent<JSX.WeRadio, HTMLWeRadioElement>('we-radio');
export const WeRadioGroup = /*@__PURE__*/createReactComponent<JSX.WeRadioGroup, HTMLWeRadioGroupElement>('we-radio-group');
export const WeSlider = /*@__PURE__*/createReactComponent<JSX.WeSlider, HTMLWeSliderElement>('we-slider');
export const WeSwitch = /*@__PURE__*/createReactComponent<JSX.WeSwitch, HTMLWeSwitchElement>('we-switch');
export const WeSwitchGroup = /*@__PURE__*/createReactComponent<JSX.WeSwitchGroup, HTMLWeSwitchGroupElement>('we-switch-group');
export const WeTab = /*@__PURE__*/createReactComponent<JSX.WeTab, HTMLWeTabElement>('we-tab');
export const WeTabGroup = /*@__PURE__*/createReactComponent<JSX.WeTabGroup, HTMLWeTabGroupElement>('we-tab-group');
export const WeTextarea = /*@__PURE__*/createReactComponent<JSX.WeTextarea, HTMLWeTextareaElement>('we-textarea');
export const WeTooltip = /*@__PURE__*/createReactComponent<JSX.WeTooltip, HTMLWeTooltipElement>('we-tooltip');
