// web-elements/dist/collection/components/chips/chips.js
import { Host, h } from "@stencil/core";
export class Chips {
    constructor() {
        this.isVisible = true;
    }
    handleRemoveChips(side) {
        if ((this.removeRight && side === 'right') || (this.removeLeft && side === 'left')) {
            this.isVisible = !this.isVisible;
            this.removeChipsCallback.emit({ value: this.value });
        }
    }
    handleSelectedChips() {
        if (this.isVisible) {
            if (this.isSelectable) {
                this.isSelected = !this.isSelected;
            }
        }
        else if (this.isSelectable) {
            this.isSelected = false;
        }
        this.selectChipsCallback.emit({ value: this.value, isSelected: this.isSelected });
    }
    render() {
        if (this.isVisible)
            return (h(Host, null,
                h("div", { class: 'chips ' + (this.isSelected && 'selected'), onClick: () => this.handleSelectedChips() },
                    this.srcImgLeft && (h("img", { class: "imageLeft", src: this.srcImgLeft, onClick: () => this.handleRemoveChips('left') })),
                    this.label && h("label", null, this.label),
                    this.srcImgRight && (h("img", { class: "imageRight", src: this.srcImgRight, onClick: () => this.handleRemoveChips('right') })))));
    }
    static get is() { return "we-chips"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() { return {
        "$": ["chips.scss"]
    }; }
    static get styleUrls() { return {
        "$": ["chips.css"]
    }; }
    static get properties() { return {
        "value": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "string",
                "resolved": "string",
                "references": {}
            },
            "required": true,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Value passed on event when chips selected or removed"
            },
            "attribute": "value",
            "reflect": false
        },
        "label": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "string",
                "resolved": "string",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Center text written inside the chips if you want a text inside the chips"
            },
            "attribute": "label",
            "reflect": false
        },
        "srcImgLeft": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "string",
                "resolved": "string",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Src of img you want to put in left side of chips if you want an image in that position"
            },
            "attribute": "src-img-left",
            "reflect": false
        },
        "srcImgRight": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "string",
                "resolved": "string",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Src of img you want to put in right side of chips if you want an image in that position"
            },
            "attribute": "src-img-right",
            "reflect": false
        },
        "removeLeft": {
            "type": "boolean",
            "mutable": false,
            "complexType": {
                "original": "boolean",
                "resolved": "boolean",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "If true the chips will be removed when user click on left image inside chips and removeChipsCallback event is triggered"
            },
            "attribute": "remove-left",
            "reflect": false
        },
        "removeRight": {
            "type": "boolean",
            "mutable": false,
            "complexType": {
                "original": "boolean",
                "resolved": "boolean",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "If true the chips will be removed when user click on right image inside chips and removeChipsCallback event is triggered"
            },
            "attribute": "remove-right",
            "reflect": false
        },
        "isSelectable": {
            "type": "boolean",
            "mutable": false,
            "complexType": {
                "original": "boolean",
                "resolved": "boolean",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "If true user can select the chips and selectChipsCallback event is triggered"
            },
            "attribute": "is-selectable",
            "reflect": false
        }
    }; }
    static get states() { return {
        "isVisible": {},
        "isSelected": {}
    }; }
    static get events() { return [{
            "method": "removeChipsCallback",
            "name": "removeChipsCallback",
            "bubbles": true,
            "cancelable": true,
            "composed": true,
            "docs": {
                "tags": [],
                "text": "Event triggered when the chips is removed"
            },
            "complexType": {
                "original": "ChipsValueRemove",
                "resolved": "ChipsValueRemove",
                "references": {
                    "ChipsValueRemove": {
                        "location": "local"
                    }
                }
            }
        }, {
            "method": "selectChipsCallback",
            "name": "selectChipsCallback",
            "bubbles": true,
            "cancelable": true,
            "composed": true,
            "docs": {
                "tags": [],
                "text": "Event triggered when the chips is selected"
            },
            "complexType": {
                "original": "ChipsValueSelect",
                "resolved": "ChipsValueSelect",
                "references": {
                    "ChipsValueSelect": {
                        "location": "local"
                    }
                }
            }
        }]; }
}
