// web-elements/dist/collection/components/radio-group/radio-group.js
import { Host, h } from "@stencil/core";
export class RadioGroup {
    constructor() {
        this.children = [];
    }
    radioCallbackHandler(prop) {
        const { value } = prop.detail;
        this.children.forEach((child) => {
            if (child['value'] == value)
                child['checked'] = true;
            else
                child['checked'] = false;
        });
        const radios = this.el.querySelectorAll(':scope > we-radio');
        radios.forEach((radio) => {
            const valueAttribute = radio.getAttribute('value');
            if (value != valueAttribute)
                radio.setAttribute('checked', 'false');
            else
                radio.setAttribute('checked', 'true');
        });
        this.radioGroupCallback.emit({ value: this.value, children: this.children });
    }
    componentDidLoad() {
        const radios = this.el.querySelectorAll(':scope > we-radio');
        radios.forEach((radio) => {
            const child = {};
            const value = radio.getAttribute('value');
            child['value'] = value;
            const checked = radio.getAttribute('checked');
            child['checked'] = checked === 'true' ? true : false;
            this.children.push(child);
        });
    }
    render() {
        return (h(Host, null,
            h("slot", null)));
    }
    static get is() { return "we-radio-group"; }
    static get encapsulation() { return "shadow"; }
    static get properties() { return {
        "value": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "string",
                "resolved": "string",
                "references": {}
            },
            "required": true,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Value that identify this radio group"
            },
            "attribute": "value",
            "reflect": false
        }
    }; }
    static get states() { return {
        "children": {}
    }; }
    static get events() { return [{
            "method": "radioGroupCallback",
            "name": "radioGroupCallback",
            "bubbles": true,
            "cancelable": true,
            "composed": true,
            "docs": {
                "tags": [],
                "text": "Event triggered when a radio button inside change its state that returning the name of radio group and the value of radio button checked"
            },
            "complexType": {
                "original": "RadioGroupValue",
                "resolved": "RadioGroupValue",
                "references": {
                    "RadioGroupValue": {
                        "location": "local"
                    }
                }
            }
        }]; }
    static get elementRef() { return "el"; }
    static get listeners() { return [{
            "name": "radioCallback",
            "method": "radioCallbackHandler",
            "target": undefined,
            "capture": false,
            "passive": false
        }]; }
}
