// web-elements/dist/cjs/we-chips.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-869cff5b.js');

const Chips = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.isVisible = true;
        this.removeChipsCallback = index.createEvent(this, "removeChipsCallback", 7);
        this.selectChipsCallback = index.createEvent(this, "selectChipsCallback", 7);
    }
    handleRemoveChips(side) {
        if ((this.removeRight && side === 'right') || (this.removeLeft && side === 'left')) {
            this.isVisible = !this.isVisible;
            this.removeChipsCallback.emit({ value: this.value });
        }
    }
    handleSelectedChips() {
        if (this.isVisible) {
            if (this.isSelectable) {
                this.isSelected = !this.isSelected;
            }
        }
        else if (this.isSelectable) {
            this.isSelected = false;
        }
        this.selectChipsCallback.emit({ value: this.value, isSelected: this.isSelected });
    }
    render() {
        if (this.isVisible)
            return (index.h(index.Host, null, index.h("div", { class: 'chips ' + (this.isSelected && 'selected'), onClick: () => this.handleSelectedChips() }, this.srcImgLeft && (index.h("img", { class: "imageLeft", src: this.srcImgLeft, onClick: () => this.handleRemoveChips('left') })), this.label && index.h("label", null, this.label), this.srcImgRight && (index.h("img", { class: "imageRight", src: this.srcImgRight, onClick: () => this.handleRemoveChips('right') })))));
    }
    static get style() { return ".chips{background:var(--background, #e5e5e5);border-radius:var(--border-radius, 0.5em);border-width:var(--border-width, 0.05em);border-color:var(--border-color, unset);width:var(--width, auto);height:var(--heigth, auto);padding:var(--padding, 0.25rem);margin:var(--margin, 0);display:var(--display, inline-block)}.chips.selected{background:var(--selected-background, #41a9c0)}.imageLeft{cursor:var(--cursor, auto);padding:var(--imageLeft--padding, 0);margin:var(--imageLeft--margin, 0 0.5rem);width:var(--imageLeft--width, auto);height:var(--imageLeft--heigth, auto);border-radius:var(--imageLeft--border-radius, 0em)}label{color:var(--color, black);font-size:var(--font-size, 12pt);font-weight:var(--font-weight, normal);text-transform:var(--text-transform, unset);text-align:var(--text-align, left);padding:var(--label--padding, 0);margin:var(--label--margin, 0 0.5rem);-webkit-text-decoration:var(--text-decoration, unset);text-decoration:var(--text-decoration, unset);pointer-events:none}.imageRight{cursor:var(--cursor, pointer);padding:var(--imageRight--padding, 0);margin:var(--imageRight--margin, 0 0.5rem);width:var(--imageRight--width, auto);height:var(--imageRight--heigth, auto);border-radius:var(--imageRight--border-radius, 0em)}"; }
};

exports.we_chips = Chips;
