// web-elements/dist/cjs/we-chips-group.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-869cff5b.js');

const ChipsGroup = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.children = [];
        this.chipsGroupCallback = index.createEvent(this, "chipsGroupCallback", 7);
    }
    removeCallbackHandler(prop) {
        const { value } = prop.detail;
        const getRemovedIndex = this.children.findIndex((item) => item.value === value);
        if (getRemovedIndex >= 0) {
            this.children.splice(getRemovedIndex, 1);
        }
        this.chipsGroupCallback.emit({ value: this.value, children: this.children });
    }
    selectCallbackHandler(prop) {
        const { value, isSelected } = prop.detail;
        this.children.forEach((child) => {
            if (child['value'] === value)
                child['isSelected'] = isSelected;
        });
        this.chipsGroupCallback.emit({ value: this.value, children: this.children });
    }
    componentDidLoad() {
        const items = this.el.querySelectorAll(':scope > we-chips');
        items.forEach((i) => {
            const child = {};
            const value = i.getAttribute('value');
            child['value'] = value;
            child['isSelected'] = false;
            this.children.push(child);
        });
    }
    render() {
        return (index.h(index.Host, null, index.h("slot", null)));
    }
    get el() { return index.getElement(this); }
    static get style() { return ":host{display:var(--display, flex);-ms-flex-pack:var(--justify-content, unset);justify-content:var(--justify-content, unset);-ms-flex-align:var(--align-items, unset);align-items:var(--align-items, unset);-ms-flex-direction:var(--flex-direction, row);flex-direction:var(--flex-direction, row)}"; }
};

exports.we_chips_group = ChipsGroup;
