// web-elements/dist/cjs/we-checkbox-group.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-869cff5b.js');

const CheckboxGroup = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.children = [];
        this.checkboxGroupCallback = index.createEvent(this, "checkboxGroupCallback", 7);
    }
    checkboxCallbackHandler(prop) {
        const { value, checked } = prop.detail;
        this.children.forEach((child) => {
            if (child['value'] == value) {
                child['checked'] = checked;
            }
        });
        this.checkboxGroupCallback.emit({ value: this.value, children: this.children });
    }
    componentDidLoad() {
        const items = this.el.querySelectorAll(':scope > we-checkbox');
        items.forEach((i) => {
            const child = {};
            const value = i.getAttribute('value');
            child['value'] = value;
            const checked = i.getAttribute('checked');
            child['checked'] = checked === 'true' ? true : false;
            this.children.push(child);
        });
    }
    render() {
        return (index.h(index.Host, null, index.h("slot", null)));
    }
    get el() { return index.getElement(this); }
};

exports.we_checkbox_group = CheckboxGroup;
