// web-elements/dist/cjs/we-radio-group.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-869cff5b.js');

const RadioGroup = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.children = [];
        this.radioGroupCallback = index.createEvent(this, "radioGroupCallback", 7);
    }
    radioCallbackHandler(prop) {
        const { value } = prop.detail;
        this.children.forEach((child) => {
            if (child['value'] == value)
                child['checked'] = true;
            else
                child['checked'] = false;
        });
        const radios = this.el.querySelectorAll(':scope > we-radio');
        radios.forEach((radio) => {
            const valueAttribute = radio.getAttribute('value');
            if (value != valueAttribute)
                radio.setAttribute('checked', 'false');
            else
                radio.setAttribute('checked', 'true');
        });
        this.radioGroupCallback.emit({ value: this.value, children: this.children });
    }
    componentDidLoad() {
        const radios = this.el.querySelectorAll(':scope > we-radio');
        radios.forEach((radio) => {
            const child = {};
            const value = radio.getAttribute('value');
            child['value'] = value;
            const checked = radio.getAttribute('checked');
            child['checked'] = checked === 'true' ? true : false;
            this.children.push(child);
        });
    }
    render() {
        return (index.h(index.Host, null, index.h("slot", null)));
    }
    get el() { return index.getElement(this); }
};

exports.we_radio_group = RadioGroup;
