// web-elements/dist/cjs/we-switch-group.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-869cff5b.js');

const SwitchGroup = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.children = [];
        this.switchGroupCallback = index.createEvent(this, "switchGroupCallback", 7);
    }
    changeSwitchCallbackHandler(prop) {
        const { value, checked } = prop.detail;
        this.children.forEach((child) => {
            if (child['value'] == value) {
                child['checked'] = checked;
            }
        });
        this.switchGroupCallback.emit({ value: this.value, children: this.children });
    }
    componentDidLoad() {
        const items = this.el.querySelectorAll(':scope > we-switch');
        items.forEach((i) => {
            const child = {};
            const value = i.getAttribute('value');
            child['value'] = value;
            const checked = i.getAttribute('checked');
            child['checked'] = checked === 'false' ? false : true;
            this.children.push(child);
        });
    }
    render() {
        return (index.h(index.Host, null, index.h("slot", null)));
    }
    get el() { return index.getElement(this); }
    static get style() { return ":host{display:var(--display, flex);-ms-flex-pack:var(--justify-content, unset);justify-content:var(--justify-content, unset);-ms-flex-align:var(--align-items, unset);align-items:var(--align-items, unset);-ms-flex-direction:var(--flex-direction, row);flex-direction:var(--flex-direction, row)}"; }
};

exports.we_switch_group = SwitchGroup;
