// web-elements/dist/cjs/we-switch.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-869cff5b.js');

const Switch = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        /** Default value of switch when component is rendered the first time */
        this.checked = false;
        this.checkedState = this.checked;
        this.switchCallback = index.createEvent(this, "switchCallback", 7);
    }
    handleChangeState() {
        this.checkedState = !this.checkedState;
        this.switchCallback.emit({ value: this.value, checked: this.checkedState });
    }
    render() {
        return (index.h(index.Host, null, index.h("input", { type: "checkbox", id: "switch", name: this.value, onChange: this.handleChangeState.bind(this), checked: this.checked }), index.h("label", { class: "switch", htmlFor: "switch" }), this.labelLeft && index.h("label", { class: "left" }, "Off"), this.labelRight && index.h("label", { class: "right" }, "On")));
    }
    static get style() { return ":host{display:block;position:relative;width:var(--container--width, 3em);height:var(--container--height, 1.5em)}:host input[type=checkbox]{display:none}:host .switch{display:block;width:100%;height:100%;position:relative;cursor:var(--cursor, pointer);background:var(--background, gray);border-radius:var(--border-radius, 6.25em)}:host .switch:after{content:\"\";position:absolute;z-index:2;top:var(--after--top, 0.22em);left:var(--after--left, 0.313em);width:var(--after--width, 1em);height:var(--after--height, 1em);background:var(--after--background, white);border-radius:var(--after--border-radius, 50%);-webkit-transition:var(--after--transition, 0.3s);transition:var(--after--transition, 0.3s)}:host input:checked+.switch{background:var(--background-checked, #3f51b5)}:host input:checked+.switch:after{left:calc(100% - 0.313em);-webkit-transform:translateX(-100%);transform:translateX(-100%)}:host .left,:host .right{position:absolute;top:50%;font-size:var(--font-size, 2em);font-weight:var(--font-weight, bold)}:host .left{left:0.625em}:host .right{right:0.625em}"; }
};

exports.we_switch = Switch;
