// web-elements/dist/cjs/we-tab-group.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-869cff5b.js');

const TabGroup = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.contentSlot = document.createElement('div');
        this.contentSlot.slot = 'tab-content';
        this.tabs = this.el.querySelectorAll('we-tab');
    }
    tabCallbackHandler(event) {
        const value = event.detail;
        const line = this.el.shadowRoot.querySelector('.line');
        this.tabs.forEach((tab, tabPosition) => {
            tab.style.width = `${100 / this.tabs.length}%`;
            if (tab.getAttribute('data-id') === value) {
                const left = tabPosition * (100 / this.tabs.length);
                const currentContent = tab.querySelector('[slot="content"]');
                this.contentSlot.innerHTML = currentContent.innerHTML;
                this.el.appendChild(this.contentSlot);
                tab.setAttribute('enabled', 'true');
                line.style.left = `${left}%`;
            }
            else {
                tab.setAttribute('enabled', 'false');
            }
        });
    }
    initTabs() {
        const line = this.el.shadowRoot.querySelector('.line');
        let getTabEnabledPosition = 0;
        this.tabs.forEach((tab, position) => {
            if (tab.getAttribute('enabled') === 'true') {
                getTabEnabledPosition = position;
            }
            tab.style.width = `${100 / this.tabs.length}%`;
        });
        const defaultContent = this.tabs[getTabEnabledPosition].querySelector('[slot="content"]');
        const left = getTabEnabledPosition * (100 / this.tabs.length);
        line.style.left = `${left}%`;
        this.contentSlot.innerHTML = defaultContent.innerHTML;
        line.style.width = `${100 / this.tabs.length}%`;
        this.el.appendChild(this.contentSlot);
    }
    componentDidRender() {
        this.initTabs();
    }
    render() {
        return (index.h(index.Host, null, index.h("header", null, index.h("slot", null), index.h("div", { class: "line" })), index.h("slot", { name: "tab-content" })));
    }
    get el() { return index.getElement(this); }
    static get style() { return ":host header{display:-ms-flexbox;display:flex;position:relative}:host header .line{position:absolute;z-index:10;height:var(--enabled-height, 0.125rem);background:var(--enabled-background-color, #3f51b5);top:var(--enabled-top, 2.125rem);left:0;-webkit-transition:0.25s ease;transition:0.25s ease}"; }
};

exports.we_tab_group = TabGroup;
